File: app/src/main/java/com/ytempest/wanandroid/di/component/BaseFragmentComponent.java
package com.ytempest.wanandroid.di.component;

import com.ytempest.wanandroid.base.fragment.MvpFragment;

import dagger.Subcomponent;
import dagger.android.AndroidInjectionModule;
import dagger.android.AndroidInjector;

/**
 * @author heqidu
 * @since 2020/6/30
 */
@Subcomponent(modules = AndroidInjectionModule.class)
public interface BaseFragmentComponent extends AndroidInjector<MvpFragment> {

    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<MvpFragment> {
    }
}

File: app/src/main/java/com/ytempest/wanandroid/di/component/BaseFragmentComponent.kt
package com.ytempest.wanandroid.di.component

import com.ytempest.wanandroid.base.fragment.MvpFragment
import dagger.Subcomponent
import dagger.android.AndroidInjectionModule
import dagger.android.AndroidInjector

/**
 * @author heqidu
 * @since 21-2-7
 */
@Subcomponent(modules = [AndroidInjectionModule::class])
interface BaseFragmentComponent : AndroidInjector<MvpFragment<*, *>> {

    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<MvpFragment<*, *>>() {
    }
}